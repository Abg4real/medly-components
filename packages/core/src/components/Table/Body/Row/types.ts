import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';
import { GridTemplateProps } from '../../types';

export type StyledProps = GridTemplateProps &
    HTMLProps<HTMLTableRowElement> &
    WithThemeProp & {
        isSelected?: boolean;
    };
export type Props = Omit<HTMLProps<HTMLTableRowElement>, 'style' | 'data'> & {
    data: { [key: string]: any };
    selectedRowIds?: number[];
    addColumnMaxSize: (field: string, value: number) => void;
    onRowSelection?: (id: number) => void;
    showShadowAfterFrozenElement?: boolean;
};
