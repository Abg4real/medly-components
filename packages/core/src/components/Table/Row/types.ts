import { HTMLProps } from '@medly-components/utils';
import { GridTemplateProps } from '../types';

export type Props = GridTemplateProps & HTMLProps<HTMLLIElement>;
