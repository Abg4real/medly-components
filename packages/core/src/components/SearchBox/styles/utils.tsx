import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
