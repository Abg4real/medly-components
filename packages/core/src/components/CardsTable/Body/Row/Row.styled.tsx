import { styled } from '@medly-components/utils';
import Card from '../../../Card';
import { Props } from './types';

export const Row = styled(Card)<Props>`
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'grab' : 'inherit')};
    box-sizing: border-box;
    border-radius: 0.8rem;
    border: ${({ withLightTheme }) => withLightTheme ? 'default' : 'none'};
    box-shadow: ${({ withLightTheme }) => withLightTheme ? 'none' : '0 0.2rem 0.8rem rgba(176,188,200,0.2)'};
        
    &:hover {
        box-shadow: ${({ withLightTheme }) => withLightTheme ? '0 0.2rem 0.4rem rgba(176,188,200,0.60)' : '0 0.4rem 0.8rem rgba(176,188,200,0.60)'};
    };

    & > * {
        background-color: ${({ theme, disabled }) => `${disabled ? theme.cardsTable.rowDisabledColor : 'inherit'}`};
    };

    :nth-child(2n) {
        background: ${({ theme, withLightTheme}) => withLightTheme ? theme.cardsTable.rowDisabledColor : theme.cardsTable.rowBackgroundColor};
    };
`;