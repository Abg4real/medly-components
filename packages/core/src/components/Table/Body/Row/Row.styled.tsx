import { css, styled } from '@medly-components/utils';
import { GridTemplateProps } from '../../types';
import { Props } from './types';

const selectedBorder = css`
    &&& {
        &::before {
            content: '';
            left: 0;
            top: -0.05rem;
            z-index: 10;
            height: 100%;
            width: 0.4rem;
            position: absolute;
            background-color: ${({ theme }) => theme.table.row.selectedBorderColor};
        }
    }
`;

export const Row = styled('tr').attrs(({ gridTemplateColumns }: GridTemplateProps) => ({
    style: {
        gridTemplateColumns
    }
}))<Props>`
    display: grid;
    position: relative;
    align-items: center;
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'pointer' : 'inherit')};

    &:nth-child(odd) {
        & > * {
            background-color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : 'odd']};
        }
    }

    &:nth-child(even) {
        & > * {
            background-color: ${({ theme, isSelected, disabled }) =>
                theme.table.row.bgColor[disabled ? 'disabled' : isSelected ? 'selected' : 'even']};
        }
    }

    &:not(:last-child) {
        & > * {
            border-bottom: 1px solid ${({ theme }) => theme.table.borderColor};
        }
    }

    ${({ isSelected }) => isSelected && selectedBorder}
`;

export const NoResult = styled(Row)`
    text-align: center;
    line-height: 30px;
`;
