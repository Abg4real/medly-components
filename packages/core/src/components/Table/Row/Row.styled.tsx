import { styled } from '@medly-components/utils';
import { GridTemplateProps } from '../types';
import { Props } from './types';

export const Row = styled('tr').attrs(({ gridTemplateColumns }: GridTemplateProps) => ({
    style: {
        gridTemplateColumns
    }
}))<Props>`
    display: grid;
    position: relative;
    align-items: center;
    cursor: ${({ disabled, onClick }) => (disabled ? 'not-allowed' : onClick ? 'pointer' : 'inherit')};

    &:nth-child(odd) {
        & > * {
            background-color: ${({ theme }) => theme.table.row.bgColor.odd};
        }
    }

    &:nth-child(even) {
        & > * {
            background-color: ${({ theme }) => theme.table.row.bgColor.even};
        }
    }

    &:not(:last-child) {
        & > * {
            border-bottom: 1px solid ${({ theme }) => theme.table.borderColor};
        }
    }
`;

export const NoResult = styled(Row)`
    text-align: center;
    line-height: 30px;
`;
