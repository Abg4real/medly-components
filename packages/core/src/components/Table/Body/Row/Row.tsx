import React, { useCallback, useContext, useState } from 'react';
import { GroupCell } from '../../GroupCell';
import { getGridTemplateColumns } from '../../helpers';
import { TablePropsContext } from '../../TableProps.context';
import { TableColumnConfig } from '../../types';
import Cell from '../Cell';
import { ExpandedCell } from '../Cell/Cell.styled';
import * as Styled from './Row.styled';
import { Props } from './types';

export const Row: React.FC<Props> = React.memo(props => {
    const [isExpanded, setExpansionState] = useState(false),
        { data, showShadowAfterFrozenElement, selectedRowIds, onRowSelection, addColumnMaxSize, ...restProps } = props,
        { columns, isLoading, onRowClick, rowIdentifier, rowClickDisableKey, rowSelectionDisableKey } = useContext(TablePropsContext);

    const handleExpansionIconClick = useCallback(() => setExpansionState(val => !val), []),
        stopPropagation = useCallback((e: React.MouseEvent) => e.stopPropagation(), []),
        getCells = (rowData: any = {}, configs: TableColumnConfig[] = columns, field = '') =>
            configs.map(config => {
                const fieldName = `${field && `${field}.`}${config.field}`;

                return config.children ? (
                    <GroupCell
                        as={field ? 'div' : 'td'}
                        key={config.field}
                        hidden={config.hidden}
                        gridTemplateColumns={getGridTemplateColumns(config.children)}
                    >
                        {getCells(rowData[config.field], config.children, config.field)}
                    </GroupCell>
                ) : (
                    <Cell
                        key={fieldName}
                        as={field ? 'div' : 'td'}
                        isLoading={isLoading}
                        isRowClickDisabled={rowData[rowClickDisableKey]}
                        isRowSelectionDisabled={rowData[rowSelectionDisableKey]}
                        data={rowData[config.field]}
                        rowId={rowData[rowIdentifier]}
                        dottedFieldName={fieldName}
                        isExpanded={isExpanded}
                        onExpansionIconClick={handleExpansionIconClick}
                        showShadowAtRight={showShadowAfterFrozenElement}
                        isRowSelected={selectedRowIds.includes(rowData[rowIdentifier])}
                        {...{ config, onRowSelection, addColumnMaxSize }}
                    />
                );
            }),
        handleRowClick = (rowData: any) =>
            !isLoading && onRowClick && !rowData[rowClickDisableKey] ? () => onRowClick(rowData) : undefined;

    return (
        <>
            <Styled.Row
                {...restProps}
                disabled={data[rowClickDisableKey]}
                onClick={handleRowClick(data)}
                isSelected={!isLoading && selectedRowIds.includes(data[rowIdentifier])}
                gridTemplateColumns={getGridTemplateColumns(columns)}
            >
                {getCells(data)}
                {isExpanded && <ExpandedCell onClick={stopPropagation}>Something</ExpandedCell>}
            </Styled.Row>
        </>
    );
});
Row.displayName = 'Row';
