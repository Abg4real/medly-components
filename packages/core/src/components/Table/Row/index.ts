import { NoResult, Row } from './Row.styled';

export default Row;

export { NoResult };
