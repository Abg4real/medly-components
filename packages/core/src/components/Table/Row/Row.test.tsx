import { TestUtils } from '@medly-components/utils';
import React from 'react';

import { NoResult, Row } from './Row.styled';

describe('Row', () => {
    it('should render row properly', () => {
        const { container } = TestUtils.render(<Row>Dummy</Row>);
        expect(container).toMatchSnapshot();
    });

    it('should render NoResult properly', () => {
        const { container } = TestUtils.render(<NoResult>Dummy</NoResult>);
        expect(container).toMatchSnapshot();
    });
});
