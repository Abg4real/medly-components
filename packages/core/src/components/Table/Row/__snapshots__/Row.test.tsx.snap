// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row should render NoResult properly 1`] = `
.c0 {
  display: grid;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: inherit;
  text-align: center;
  line-height: 30px;
}

.c0:nth-child(odd) > * {
  background-color: #F7F8F9;
}

.c0:nth-child(even) > * {
  background-color: #ffffff;
}

.c0:not(:last-child) > * {
  border-bottom: 1px solid #C7D0D8;
}

<div>
  <tr
    class="sc-AxjAm c0"
  >
    Dummy
  </tr>
</div>
`;

exports[`Row should render row properly 1`] = `
.c0 {
  display: grid;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: inherit;
}

.c0:nth-child(odd) > * {
  background-color: #F7F8F9;
}

.c0:nth-child(even) > * {
  background-color: #ffffff;
}

.c0:not(:last-child) > * {
  border-bottom: 1px solid #C7D0D8;
}

<div>
  <tr
    class="c0"
  >
    Dummy
  </tr>
</div>
`;
